### PR TITLE
add support for building with jack

### PR DESCRIPTION
**Pull request by slhck**

### CODE REVIEW

**Comment by louisfoster** [comment]
Is there a reason for not merging this?

**Comment by slhck** [comment]
Yes, it does (did?) not work due to https://github.com/Homebrew/linuxbrew-core/issues/11710 which was never resolved. I have not had time to look into it again.

**Comment by carlocab** [comment]
You should be able to add `jack` to your `PKG_CONFIG_PATH` in the formula with something like
```ruby
ENV.prepend_path "PKG_CONFIG_PATH", Formula["jack"].opt_lib/"pkgconfig"
```

**Comment by slhck** [comment]
Thanks for the suggestion. I haven't touched this in a while but will try again soon.

**Review by TOMOYA-githu — APPROVED** [review]

File: Formula/ffmpeg.rb [hunk 1]
```diff
@@ -160,6 +160,12 @@ def install
       args << "--extra-ldflags=-L#{HOMEBREW_PREFIX}/include"
     end
 
+    if build.with? "jack"
+      ENV.prepend_path "PKG_CONFIG_PATH", Formula["jack"].opt_lib/"pkgconfig"
```
Comment (carlocab) [inline Formula/ffmpeg.rb:164]: If this doesn't work, try:

```suggestion
      ENV.prepend_path "PKG_CONFIG_PATH", Formula["jack"].opt_prefix/"lib64/pkgconfig"
```

Comment (slhck) [reply Formula/ffmpeg.rb:164]: The original line seems to have worked for me under Linux:

```
➜ /home/linuxbrew/.linuxbrew/bin/ffmpeg -devices | grep jack
ffmpeg version 4.4 Copyright (c) 2000-2021 the FFmpeg developers
  built with gcc 11.2.0 (Homebrew GCC 11.2.0)
  configuration: --prefix=/home/linuxbrew/.linuxbrew/Cellar/ffmpeg/4.4-with-options_1 --enable-shared --cc=gcc-11 --host-cflags= --host-ldflags= --enable-gpl --enable-libaom --enable-libdav1d --enable-libmp3lame --enable-libopus --enable-libsnappy --enable-libtheora --enable-libvorbis --enable-libvpx --enable-libx264 --enable-libx265 --enable-libfontconfig --enable-libfreetype --enable-frei0r --enable-libass --enable-demuxer=dash --disable-htmlpages --enable-libjack --enable-indev=jack
  libavutil      56. 70.100 / 56. 70.100
  libavcodec     58.134.100 / 58.134.100
  libavformat    58. 76.100 / 58. 76.100
  libavdevice    58. 13.100 / 58. 13.100
  libavfilter     7.110.100 /  7.110.100
  libswscale      5.  9.100 /  5.  9.100
  libswresample   3.  9.100 /  3.  9.100
  libpostproc    55.  9.100 / 55.  9.100
 D  jack            JACK Audio Connection Kit
```

Comment (carlocab) [reply Formula/ffmpeg.rb:164]: Great; glad you're set then.

Comment (slhck) [reply Formula/ffmpeg.rb:164]: Thanks for your help implementing this!

**Comment by slhck** [comment]
@retokromer I would appreciate a quick review. It builds both on macOS 11 and Ubuntu Linux 20.04 for me.

**Review by retokromer — APPROVED** [review]
It builds on macOS 10.15.7 and 10.14.6, Debian 11.0 and Ubuntu 21.04 on my end.

**Pull request merged**